aiidalab/app.py

```diff
@@ -90,6 +92,17 @@ class AiidaLabApp(traitlets.HasTraits):
     busy = traitlets.Bool(readonly=True)
     modified = traitlets.Bool(readonly=True, allow_none=True)
 
+    @dataclass
+    class AppRegistryData:
+        """Dataclass that contains the app data from the app registry."""
+        git_url: str
+        meta_url: str
+        categories: List[str]
+        groups: List[str]  # appears to be a duplicate of categories?
```

Comment:
@CasperWA Do you know what's up with that?

Reply:
I don't know what `groups` refer to. But it may be something @yakutovicha introduced, either for internal use or not.
The official keys for the registry can be found in the README of the [`aiidalab-registry`](https://github.com/aiidalab/aiidalab-registry) repository. Here only `categories` appears.

Reply:
So it seems that `groups` is the key to the `categories` definitions in the app registry file, whereas `categories` contains the Category name. I think we can keep both for now, but we should probably make sure to harmonize that at some point.

Reply:
Right. So one is the "look up" and the other is the values for any given app? Or did I misunderstand?

Reply:
No, the categories are actually defined in the metadata registry as part of the top-level `categories` group. The `apps/<app-id>/groups` values match the `categories/<category-id>` keys. The `apps/<app-id>/categories` values actually contain the formatted name of the categories. This means that the categories of apps are over-specified.

Reply:
So should we try to focus the information in this PR as well, or do you want to leave it for another go at it?

Reply:
This data class is currently reflecting the information that is provided to the class. That information is partially redundant, but I'd prefer to address that issue in a separate change set.

Reply:
For the record: `groups` is an exact copy of `categories`. It was keept for the backwards compatibility. It is created [here](https://github.com/aiidalab/ansible-role-aiidalab-server/blob/9057f349f2a4fe5372628390d79a179079af60e5/files/merge_app_metadata.py#L109-L110)